src/input_device/JoystickPool.cpp

```diff
@@ -7,22 +7,24 @@
 
 #include <SDL2/SDL.h>
 
-// Joystick Pool
-JoystickPool* JoystickPool::mSingleton = nullptr; //static
+#include <InputManager.h>
 
-JoystickPool& JoystickPool::getSingleton()
+// Joystick Pool
+JoystickPool::~JoystickPool()
 {
-	if (mSingleton == nullptr)
-		mSingleton = new JoystickPool();
-
-	return *mSingleton;
+	for (auto& iter : mJoyMap)
+	{
+		SDL_JoystickClose(iter.second);
+	}
+	mJoyMap.clear();
```

Comment:
the call to `clear` is superfluous here, I think, since we're in the destructor anyway.